test/python/circuit/test_controlled_gate.py

```diff
@@ -667,6 +667,76 @@ def test_open_controlled_unitary_matrix(self, num_ctrl_qubits):
                          (output_target != cond_output)) or
                         output_ctrl != input_ctrl)
 
+    def test_open_control_cx_unrolling(self):
+        """test unrolling of open control gates when gate is in basis"""
+        qc = QuantumCircuit(2)
+        qc.cx(0, 1, ctrl_state=0)
+        dag = circuit_to_dag(qc)
+        unroller = Unroller(['u3', 'cx'])
+        uqc = dag_to_circuit(unroller.run(dag))
+        self.assertEqual(len(uqc.data), 3)
```

Comment:
One general question: what's the benefit of testing the DAG contents instead of comparing the circuit to a reference circuit? In principle comparing to a reference circuit seems easier to me, especially for such a small one.

Reply:
Practically, there's no difference. Since the unroller returns a DAG, comparing to a DAG is a little more convenient. (`QuantumCircuit.__eq__` actually does two DAG conversions, and then does an `is_isomorphic` test.)

Reply:
I'll update it.